### PR TITLE
Make amp-carousel always show controls in emails

### LINKED ISSUE #25713: Make amp-carousel[controls] the default behavior in AMP for Email
## Describe the new feature or change to an existing feature you'd like to see

Make `<amp-carousel>` work the same as `<amp-carousel controls>` for AMP for Email.

## Describe alternatives you've considered

Alternative: make `controls` mandatory for `amp-carousel` in the validator. This is not really ideal developer experience.

## Additional context

Gmail on mobile has the swiping motion reserved for moving between emails and thus [documents](https://developers.google.com/gmail/ampemail/tips) that users should always use `controls` on mobile. It seems many other mobile clients have the same problem.

### PR DESCRIPTION
**Pull request by fstanis**
Fixes #25713

### CODE REVIEW

**Comment by sparhami** [comment]
I feel like this should be done in a generic way, at some level asking the viewer if they support swipe. There are other components that may need the same treatment. Another option is to enforce the usage of the `controls` attribute via the validator, with a separate spec that requires it for the email format.

/cc @dvoytenko

**Comment by fstanis** [comment]
> I feel like this should be done in a generic way, at some level asking the viewer if they support swipe.

I like this. Maybe even a doc-level parameter that overrides [`isMouseDetected`](https://github.com/ampproject/amphtml/blob/d824612e0665eea8165cac00eb6d210ec81717ab/src/input.js#L135-L137)? For example, `https://amp.example/page#mouse=1` forces that function to return true while `https://amp.example/page#mouse=0` forces it to be false?

This not just solves the issue for email (since email clients control the viewer and can choose to override this), but also helps test mouse-related functionality for developers.

> Another option is to enforce the usage of the `controls` attribute via the validator, with a separate spec that requires it for the email format.

This was my first idea, but it breaks existing emails and templates and puts needless burden on developers. I'd rather we solve this on runtime or viewer level if at all possible.

**Comment by dvoytenko** [comment]
> I feel like this should be done in a generic way, at some level asking the viewer if they support swipe.

I'm pretty sure that has already been done at some point before. For instance, don't we always show carousel controls on desktop?

**Comment by sparhami** [comment]
> > I feel like this should be done in a generic way, at some level asking the viewer if they support swipe.
> 
> I'm pretty sure that has already been done at some point before. For instance, don't we always show carousel controls on desktop?

Carousel 0.1 uses `.amp-mode-mouse`, but I'm not sure whether adding this class in emails for mobile devices would be the right thing to do. I think we should treat it as `@media (pointer: fine)`. Carousel 0.2 shows the controls until the user changes slides specifically with a touch.

It seems like carousel is the only AMP-internal thing using `.amp-mode-mouse`, but it seems like sites use it in their CSS. Since email is new, it may be safe to add it in the email format, but I think having a separate mechanism would be preferable.

**Comment by fstanis** [comment]
> For instance, don't we always show carousel controls on desktop?

They don't show on my Linux laptop that has a touchscreen.

> but I'm not sure whether adding this class in emails for mobile devices would be the right thing to do

This isn't sufficient I think - I just checked, my laptop has both `amp-mode-touch amp-mode-mouse` and the controls don't show.

Now that Gmail is rolling out mobile support, I'd merge this PR as a temporary solution for #25713. I'm happy to revert it when we have a better way to do it, e.g. via a viewer capability, but I'd like to avoid carousels breaking on mobile as more users start getting AMP emails there and we haven't communicated proactively enough about this feature.

File: extensions/amp-carousel/0.2/amp-carousel.js [hunk 1]
```diff
@@ -467,8 +468,11 @@ class AmpCarousel extends AMP.BaseElement {
    */
   updateUi_() {
     const index = this.carousel_.getCurrentIndex();
+    const doc = /** @type {!Document} */ (this.element.ownerDocument);
     const bothDisabled =
-      this.hadTouch_ && !this.element.hasAttribute('controls');
+      !isAmp4Email(doc) &&
+      this.hadTouch_ &&
+      !this.element.hasAttribute('controls');
```
Comment (choumx) [inline extensions/amp-carousel/0.2/amp-carousel.js]: Let's revert these 0.2 changes.

Comment (fstanis) [reply extensions/amp-carousel/0.2/amp-carousel.js]: Done.

**Pull request merged**